doc/user/layouts/shortcodes/network-security/privatelink-mysql.md

```diff
@@ -0,0 +1,147 @@
+
+    **Remarks**:
+
+    a. Network Load Balancers do not have associated security groups. Therefore, the security groups for your targets must use IP addresses to allow traffic.
+
+    b. You can't use the security groups for the clients as a source in the security groups for the targets. Therefore, the security groups for your targets must use the IP addresses of the clients to allow traffic. For more details, check the [AWS documentation](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-register-targets.html).
+
+1. #### Create a Network Load Balancer (NLB)
+    Create a [Network Load Balancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-network-load-balancer.html) that is **enabled for the same subnets** that the RDS instance is in.
+
+1. #### Create TCP listeners
+
+    Create a [TCP listener](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-listener.html) for your RDS instance that forwards to the corresponding target group you created.
+
+1. #### Create a VPC endpoint service
+
+    Create a VPC [endpoint service](https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html) and associate it with the **Network Load Balancer** that you’ve just created.
+
+    Note the **service name** that is generated for the endpoint service.
+
+1. #### Create an AWS PrivateLink connection
+
+    In Materialize, create an [AWS PrivateLink connection](/sql/create-connection/#aws-privatelink)
+    that references the endpoint service that you created in the previous step.
+
+    **Availability zone selection**:
+
+    * For **in-region connections**, the availability zones of the NLB and the
+      consumer VPC **must match**.
+
+        ```mzsql
+        CREATE CONNECTION privatelink_svc TO AWS PRIVATELINK (
+            SERVICE NAME 'com.amazonaws.vpce.<region_id>.vpce-svc-<endpoint_service_id>',
+            -- Update the list of the availability zones to match the ones in
+            -- your AWS account.
+            AVAILABILITY ZONES ('use1-az1', 'use1-az2', 'use1-az4')
+        );
+        ```
+
+    * For **cross-region connections** only, it is recommended to specify no availability zones.
+
+        ```mzsql
+        CREATE CONNECTION privatelink_svc TO AWS PRIVATELINK (
+        SERVICE NAME 'com.amazonaws.vpce.us-west-1.vpce-svc-<endpoint_service_id>',
+        AVAILABILITY ZONES ()
+        );
+        ```
+
+        The `SERVICE NAME` region refers to where the endpoint service was
+        created. In this example, `AVAILABILITY ZONES` will be optimally
+        auto-assigned if none are provided. This setup allows you to **connect
+        to the endpoint service from a different region** than the one where
+        your Materialize instance is running.
+
+    * For **cross-region connections**, you can select any availability zones in your current
```

Comment:
@jubrad, just double-checking: I think we want to get rid of this paragraph, right? It's in conflict with the above. For cross-region connections, we'll recommend _not_ specifying any AZs, with a note saying the clause will be made optional in a future release.

No need to make changes, I'll work them in!

Reply:
Rephrased and restructured this across all documentation pages.